src/printer.js

```diff
@@ -1209,13 +1209,18 @@ function genericPrintNoParens(path, options, print) {
       else parts.push("default:");
 
       if (n.consequent.find(node => node.type !== "EmptyStatement")) {
-        const cons = path.call(
-          function(consequentPath) {
-            return printStatementSequence(consequentPath, options, print);
-          },
-          "consequent"
-        );
-
+        const cons = path.call(consequentPath => {
+          let printed = [];
+          path.map(p => {
+            const parentParent = path.getParentNode(1);
```

Comment:
`parentParent` and `last` do not change on every iteration, you can put them outside of the loop